src/app/portlets/dot-templates/guards/dot-template.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { CanLoad, Route, UrlSegment } from '@angular/router';
+import { DotRouterService } from '@services/dot-router/dot-router.service';
+
+@Injectable()
+export class DotTemplateGuard implements CanLoad {
+    constructor(private dotRouterService: DotRouterService) {}
+
+    canLoad(_route: Route, segments: UrlSegment[]): boolean {
+        const { path } = segments[1];
+
+        if (path === 'designer' || path === 'advanced') {
+            return true;
+        }
+
+        this.dotRouterService.gotoPortlet('templates');
+
+        return false;
```

Comment:
How about?

```javascript
canLoad(_route: Route, segments: UrlSegment[]): boolean {
    const { path } = segments[1];
    this.dotRouterService.gotoPortlet('templates');
    const isValidTemplatePath = path === "designer" || path === "advanced";

    if(!isValidTemplatePath) {
      this.dotRouterService.gotoPortlet('templates');
      return false;
    }

    return isValidTemplatePath;
}
```

Reply:
Can't do The `gotoPortlet` is only if is `false`

But I like the `isValidTemplatePath` so I use that